### PR TITLE
List vs Arraylist (Equals)

### PR DESCRIPTION
**Pull request by SavvaSpb**

### CODE REVIEW

File: Console_Arraylist/Program.cs [hunk 1]
```diff
@@ -50,15 +50,144 @@ static void Main(string[] args)
...
+            //        element = employeesList[i];
+            //}
+
+            //Console.WriteLine("\n MyList contains {0} \n", element);
+
+            ////////////////////////////////////////////////////////////////////////////////////////
+
+
+
+            List<Person> test = new List<Person>();
```
Comment (TigranDataArt) [inline Console_Arraylist/Program.cs]: please rename test to personsList

**Review by TigranDataArt — CHANGES_REQUESTED** [review]

File: Console_Arraylist/Program.cs (hunk 1 above)
Comment (SavvaSpb) [reply Console_Arraylist/Program.cs]: Yes , its looks better. Already renamed test to personsList .

File: Console_Arraylist/MyArrayList.cs [hunk 2]
```diff
@@ -16,17 +16,22 @@ public class MyArrayList : IList
...
-                _items = emptyarray;
-            else
-                _items = new object[capacity];
+            //if (capacity < 0)
+            //    throw new ArgumentOutOfRangeException("capacity");
+
+            //if (capacity == 0)
+            //    _items = emptyarray;
+            //else
+            //    _items = new object[capacity];
```
Comment (TigranDataArt) [inline Console_Arraylist/MyArrayList.cs]: Please remove not needed comments after refactoring.

File: Console_Arraylist/MyArrayList.cs [hunk 3]
```diff
@@ -16,17 +16,22 @@ public class MyArrayList : IList
         private const int _defaultCapacity = 4;
         private object[] emptyarray = new object[0];
 
-        public MyArrayList(int capacity)
+        public MyArrayList()
```
Comment (TigranDataArt) [inline Console_Arraylist/MyArrayList.cs:19]: Why you replaced constructor with empty one, It's better to have both constructors one with capacity, the other one without capacity. Please refactor code to have both constructors.

File: Console_Arraylist/Person.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+namespace Console_Arraylist
+{
+    public class Person : IEquatable<Person>
+    {
+        public int PasportId { get; set; }
+
+        public string Name { get; set; }
+
+        //public string FirstName { get; set; }
```
Comment (TigranDataArt) [inline Console_Arraylist/Person.cs]: Why you commented FirstName ? If you don't need It just delete It.

File: Console_Arraylist/Person.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+
+            //bool result = obj is Person person && person.GetHashCode() == GetHashCode();
+            //return result;
+
+
+
+            //if (obj is Person person) 
+            //    return PasportId == person.PasportId;
+            //return false;
```
Comment (TigranDataArt) [inline Console_Arraylist/Person.cs]: Please remove unused and commented code.

**Review by TigranDataArt — CHANGES_REQUESTED** [review]

**Review by TigranDataArt — APPROVED** [review]

**Pull request merged**